### PR TITLE
kubeadm: use the kinder/update-workflows tool to generate specs

### PR DESCRIPTION
**Pull request by neolit123**
The kubeadm project is adding tooling to automatically update
Prow job specs based on templates.

Run the same tool on the existing set of Prow jobs.

xref https://github.com/kubernetes/kubeadm/pull/2434
fixes https://github.com/kubernetes/kubernetes/issues/101231

### CODE REVIEW

**Comment by neolit123** [comment]
/cc @pacoxu @RA489 
looking for LGTM

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kubelet-x-on-y.yaml [hunk 1]
```diff
@@ -41,15 +41,15 @@ periodics:
           memory: "9000Mi"
           cpu: 2000m
 
-- name: ci-kubernetes-e2e-kubeadm-kinder-kubelet-1-20-on-1-21
+- name: ci-kubernetes-e2e-kubeadm-kinder-kubelet-1-20-on-latest
```
Comment (fabriziopandini) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kubelet-x-on-y.yaml:44]: I'm a little bit confuse by this test being changed by regeneration..
Also should this be 1.21-on-latest?

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+
 - name: ci-kubernetes-e2e-kubeadm-kinder-upgrade-1-21-latest
   interval: 2h
   decorate: true
   labels:
     preset-dind-enabled: "true"
     preset-kind-volume-mounts: "true"
   annotations:
-    testgrid-dashboards: sig-cluster-lifecycle-kubeadm,sig-release-master-informing
+    testgrid-dashboards: sig-cluster-lifecycle-kubeadm, sig-release-master-informing
```
Comment (fabriziopandini) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml:11]: Just for double checking, Is the additional space here tolerated by test grid machinery?

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kubelet-x-on-y.yaml (hunk 1 above)
Comment (fabriziopandini) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kubelet-x-on-y.yaml:44]: Ohh might be this is actually a new test and the old one is now down in the file...

**Review by fabriziopandini — COMMENTED** [review]
Overall LGTM, two questions

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kubelet-x-on-y.yaml (hunk 1 above)
Comment (ehashman) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kubelet-x-on-y.yaml:44]: Yup this is adding a new skew case.

**Review by ehashman — COMMENTED** [review]
LGTM but I don't know the answer to the questions

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kubelet-x-on-y.yaml (hunk 1 above)
Comment (neolit123) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kubelet-x-on-y.yaml:44]: the diffs are noisy and difficult to review. thus the machinery to update them.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml (hunk 2 above)
Comment (neolit123) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml:11]: yes

**Comment by neolit123** [comment]
/hold to sync with the k/kubeadm preparation
https://github.com/kubernetes/kubeadm/pull/2434

**Comment by pacoxu** [comment]
`ci auto update PR` #21850 make this conflicting.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kubelet-x-on-y.yaml (hunk 1 above)
Comment (pacoxu) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kubelet-x-on-y.yaml:44]: the diffs depend on the order.

`ci-kubernetes-e2e-kubeadm-kinder-kubelet-1-20-on-1-21` below(or next) the old case
`ci-kubernetes-e2e-kubeadm-kinder-kubelet-1-20-on-latest` here is a new case

/lgtm
This Pr may need LGTM after rebase later.

**Comment by fabriziopandini** [comment]
ready to lgtm after rebase

**Comment by neolit123** [comment]
@fabriziopandini @pacoxu 
rebased, looking for LGTM again.

**Review by ehashman — COMMENTED** [review]
/lgtm

**Pull request merged**